### PR TITLE
search: show tab keyboard tip on MonacoQueryInput autocomplete suggestions

### PR DESCRIPTION
**Pull request by limitedmage**
There seems to be no way to hook into the Monaco Editor suggestions UI through the API, so I came up with a CSS-only fix that works nicely.

Would love some feedback from @quinnkeast or @rrhyne on the design and color choices.

image

image

### CODE REVIEW

**Comment by quinnkeast** [comment]
This moved so quickly! Sorry that you didn't have the Figma file for reference already. Here's the specs for the badge from design:

```
border-color: $color-border-1;
color: $color-text-3;
font-size: 0.65rem;
line-height: 1;
padding: 0.25rem 0.375rem;
```
As well, can this conditionally only be shown when there's no information icon `( i )`? I'm worried that when shown together, the result row is getting a bit too full.

Figma for reference: https://www.figma.com/file/cVAsl1QK5Voox0cwvySoJV/Tab-hint-Experiment?node-id=282%3A260

**Comment by rrhyne** [comment]
I think it's better that it displays alongside the information icon. Hiding it while the action is still required for selection is inconsistent.

**Comment by limitedmage** [comment]
Thanks @quinnkeast! This looks great in light mode. Do you have designs for dark mode? `$color-border` seems to be the same (or very similar) color to the background and `$color-text-3` is white.

I agree with @rrhyne that we should always show the tab hint to be consistent.

image

image

**Comment by felixfbecker** [comment]
**suggestion**: Is there a way we could make it aligned with our [`<kbd>` style](https://5f0f381c0e50750022dc6bf7-fxmwmkrlsi.chromatic.com/?path=/story/web-global-styles--code) used for keyboard shortcuts everywhere else? I.e. either duplicating the code or even using Sass `@extend` or a mixin?

**Comment by limitedmage** [comment]
@felixfbecker Love this idea!! Didn't know about our `kbd` styling. I implemented it but had to change the background color in light mode because it was the same color as the suggestions list. Here's how it looks:

image

image

File: web/src/search/input/MonacoQueryInput.scss [hunk 1]
```diff
@@ -41,6 +41,17 @@
         .inputarea.focus-visible {
             box-shadow: none;
         }
+
+        // Show keyboard tip to press Tab to complete
+        .suggest-widget .monaco-list .monaco-list-row.focused::after {
+            @extend kbd;
```
Comment (felixfbecker) [inline web/src/search/input/MonacoQueryInput.scss:47]: Perfect ✨

**Comment by quinnkeast** [comment]
Extending the `kbd` style is good for consistency, but I'm not sure it's a good final outcome here. In documentation and text blocks, the `kbd` elements are intended to stand out to make them easier to interpret. In the search results, this goes against our goal—it's meant to be a hint, rather than a focal point. With the default `kbd` style, it becomes more of a focal point than the content of the search result, which introduces a new usability issue.

Can we instead `@extend` with the `kbd` style but then override the `box-shadow`, `border-color`, and `font-size`attributes?

**Comment by felixfbecker** [comment]
I see your point. I'd say though that the goal of the `kbd` style is not to _stand out_, but to communicate that this is a keyboard shortcut, that's why it's styled to look like a key on a keyboard (with a bit of depth). So that is the visual aspect I'd want to keep. To solve your concern in an easy way, maybe we can just give it a reduced `opacity`?

**Comment by quinnkeast** [comment]
I agree with your thinking around the purpose of the `kbd` style really emphasizing that it's a keyboard shortcut. Still, the depth effect inherently causes it to have a greater visual prominence than the rest of the line. In most contexts, I'd be comfortable with it, but in this specific context I'm really concerned about the unintentional impact. I doubt our users will have any trouble recognizing it as a shortcut hint even without the depth effect.

**Comment by limitedmage** [comment]
I agree with @quinnkeast that this is meant to be a subtle hint, and also because we don't use `<kbd>` very much at all on the search site (it's mostly used in documentation), so I think having a different styling would be OK. These are the final colors I've settled on:

image

image

**Comment by quinnkeast** [comment]
Looks great! Looking forward to having this.

**Pull request merged**